cartridge/shop/views.py

```diff
@@ -282,6 +299,15 @@ def checkout_steps(request, form_class=OrderForm, extra_context=None):
                 # and send the order receipt email.
                 order = form.save(commit=False)
                 order.setup(request)
+
+                if (not settings.SHOP_PAYMENT_STEP_ENABLED or
+                        order.total == 0):
```

Comment:
I haven't tried it but that makes sense. `SHOP_PAYMENT_STEP_ENABLED` seems to only effect the template context. But now that you've added `step += 1`, neither of these conditions should be possible, right?